aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -102,15 +122,21 @@ struct DispatchKeyExtractor final {
     return impl::dispatchTypeId(typeSet);
   }
 
-  explicit DispatchKeyExtractor(size_t num_args)
-  : num_args_(num_args) {}
+  explicit DispatchKeyExtractor(
+    size_t num_args,
+    c10::optional<size_t> scalar_type_index)
+  : num_args_(num_args),
+    scalar_type_index_(scalar_type_index) {}
 
   // this is caching the index so we don't have to parse the schema inputs
   // again and again for each dispatcher lookup.
   // num_args_ is allowed to be zero; that just means you must do the
   // fallthrough
   // TODO: a potential optimization is to store a bitfield of arg locations,
   size_t num_args_;
+  // this is caching the index for ScalarType for args inside
```

Comment:
Maybe, in the final version also add a bit more comment here, explaining that if this is set, then the argument at this index must be ScalarType, at index+1 and index+2 must be ... and if this is not set, then we know that the op does not have this constellation of args